### ISSUE TITLE
some confusions

### ISSUE THREAD

**Issue by leileilin** [TharinduDR/TransQuest#27]
Hey man. 
I would like to ask you whether the fine tuning of your pre training model is to fine tune the parameters of the whole pre training model together?

Because I built a QE model by using the pre training language model, but the hter output of a batch during validation is always the same value. I don't know why?
thany you man.

**Comment by TharinduDR** [comment]
What is the training process you used?

**Comment by leileilin** [comment]
> What is the training process you used?

Here are my code for training process .

self.bert.train()
for i, data in train_data_iter:
    # 0. batch_data will be sent into the device(GPU or cpu)
    data = {key: value.to(self.device) for key, value in data.items()}

    # 1. forward the next_sentence_prediction and masked_lm model
    sent_outputs, word_outputs = self.bert.forward(input_ids=data["input_ids"], attention_mask=data["attention_mask"], token_type_ids=data["token_type_ids"])

    # 2. NLLLoss of predicting masked token word
    sent_loss = self.sent_criterion(sent_outputs.squeeze(1), data["hter"])
    word_loss = self.word_criterion(word_outputs.transpose(1, 2), data["OK_BAD"])
    loss = sent_loss + word_loss

    self.optim_schedule.zero_grad()

    loss.backward()

    # Clip the norm of the gradients to 1.0 to prevent "exploding gradients"
    torch.nn.utils.clip_grad_norm_(self.bert.parameters(), 1.0)

    # 3. backward and optimization only in train
    self.optim_schedule.step_and_update_lr()

    # record training loss
    train_losses.append(loss.item())

    train_avg_loss += loss.item()

    post_fix = {
        "epoch": epoch,
        "iter": i,
        "avg_loss": train_avg_loss / (i + 1),
        "loss": loss.item()
    }

    if i % self.log_freq == 0:
        train_data_iter.write(str(post_fix))

self.bert.eval()

with torch.no_grad():

    for i, data in valid_data_iter:

        # 0. batch_data will be sent into the device(GPU or cpu)
        data = {key: value.to(self.device) for key, value in data.items()}
        
        print("input_ids: ", data["input_ids"])
        # 1. forward the next_sentence_prediction and masked_lm model
        sent_outputs, word_outputs = self.bert.forward(input_ids=data["input_ids"], attention_mask=data["attention_mask"], token_type_ids=data["token_type_ids"])

        print("sent_outputs: ", sent_outputs.squeeze(1))
        print("sent_true: ", data["hter"])

        # 2. NLLLoss of predicting masked token word
        sent_loss = self.sent_criterion(sent_outputs.squeeze(1), data["hter"])
        word_loss = self.word_criterion(word_outputs.transpose(1, 2), data["OK_BAD"])
        loss = sent_loss + word_loss

**Comment by TharinduDR** [comment]
Is this TransQuest?

**Comment by leileilin** [comment]
> Is this TransQuest?

sorry, it is my custom model.

**Comment by TharinduDR** [comment]
With TransQuest, we finetune all the parameters

**Comment by zhengxxn** [comment]
I got the same bug when running with fairseq, and I found that fairseq uses left-pad by default, like '<pad> <cls> ...', so it actually applies <pad> for prediction and hence results in the same output  ... 

I hope it will be helpful for someone likes me ...
